core/common/src/main/java/alluxio/collections/LockPool.java

```diff
@@ -256,7 +269,9 @@ private Resource getResource(K key) {
         v.mIsAccessed = true;
         return v;
       }
-      return new Resource(mDefaultLoader.apply(k));
+      Resource res = mResueLockPool.acquireWithoutBlocking();
```

Comment:
Once we change the `acquireWithoutBlocking` based on my suggestion, we should have it here:

```java
Resource res = mResueLockPool.acquireWithoutBlocking();
if (res == null) {
  res = mDefaultLoader.get();
}

Reply:
Done